mithril-aggregator/src/services/signature_processor.rs

```diff
@@ -0,0 +1,203 @@
+use std::sync::Arc;
+
+use slog::{error, warn, Logger};
+
+use mithril_common::{logging::LoggerExtensions, StdResult};
+use tokio::{select, sync::watch::Receiver};
+
+use super::{CertifierService, SignatureConsumer};
+
+/// A signature processor which receives signature and processes them.
+#[cfg_attr(test, mockall::automock)]
+#[async_trait::async_trait]
+pub trait SignatureProcessor: Sync + Send {
```

Comment:
Is this trait really useful ?

It's only added value is the ability to mock but:
- There's no use of the mock right now
- The only future use I can envision is in integration tests were it would make more sense to use the `SequentialSignatureProcessor` directly and mock/implement manually a `SignatureConsumer` instead.

Reply:
I guess it is. It can also be used to create decorators (e.g. which may be useful to record metrics) and it allows to alternative implementation of the processor (e.g. which could take more advantage of concurrency). 

I also believe this is good practice to represent business code with traits. But if we think in the long run that it is not necessary we can get rid of it :+1: